pkg/workloadmeta/types.go

```diff
@@ -0,0 +1,161 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import "time"
+
+// Kind is the kind of an entity.
+type Kind string
+
+// ContainerRuntime is the container runtime used by a container.
+type ContainerRuntime string
+
+// ECSLaunchType is the launch type of an ECS task.
+type ECSLaunchType string
+
+// EventType is the type of an event.
+type EventType int
+
+// List of enumerable constants for the types above.
+const (
+	KindContainer     Kind = "container"
+	KindKubernetesPod Kind = "kubernetes_pod"
+	KindECSTask       Kind = "ecs_task"
+
+	ContainerRuntimeDocker ContainerRuntime = "docker"
+
+	ECSLaunchTypeEC2      ECSLaunchType = "ec2"
+	ECSLaunchTypeFargate  ECSLaunchType = "fargate"
+	ECSLaunchTypeExternal ECSLaunchType = "external"
+
+	EventTypeSet EventType = iota
+	EventTypeUnset
+)
+
+// Entity is an item in the metadata store. It exists as an interface to avoid
+// usage of interface{}.
+type Entity interface {
+	GetID() EntityID
+}
+
+// EntityID represents the ID of an Entity.
+type EntityID struct {
+	Kind Kind
+	ID   string
+}
+
+// GetID satisfies the Entity interface for EntityID to allow a standalone
+// EntityID to be passed in events of type EventTypeUnset without the need to
+// build a full, concrete entity.
+func (i EntityID) GetID() EntityID {
+	return i
+}
+
+var _ Entity = EntityID{}
```

Comment:
What is this for?

Reply:
It's a poor man's "interface assertion". It ensures that if a type does not implement an interface, the compilation error will point to this line (close to the definition of the type) instead of wherever a value of that type is used as an Entity. Very useful during development.